core/src/nodes/raw_swarm.rs

```diff
@@ -152,7 +168,9 @@ where
     /// This can include, for example, an error during the handshake of the encryption layer, or
     /// the connection unexpectedly closed.
     IncomingConnectionError {
-        /// Address of the listener which received the connection.
+        /// ID of the listener which received the connection.
+        listener_id: Id,
```

Comment:
Having a `listener_id` is only necessary if we want to allow closing a listener, which isn't possible at the moment.
On one hand it's nice to have the tools to allow that later, on the other hand we might never add this feature or add it in a different way (eg. allow closing individual multiaddresses or something).
IMO we should just not have a `listener_id`, and we can always add it if we want to allow cancelling a listener entirely.

Reply:
The listener ID was introduced because previously the listener address was also used for identification purposes, e.g. in tests. It removed the ID now. Without being able to refer to individual listeners, some tests now made no longer sense and had to be removed as well.